### PR TITLE
Keep row selected when changing dateRange.

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue #2644.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Review by felixarntz — CHANGES_REQUESTED** [review]
The existing code looks good, but there's one problem this introduces: We need to make sure that `selectedRow` is no longer set when changing the tab (i.e. the `dimensionName`). Right now, when I select a pie slice in e.g. "Channels" and then go over to "Locations", there will still be a slice initially selected (I guess the one with the same row index or something?). That should not be the case, the selection should only persist within the same dimension's pie chart.

We should reset `selectedRow` when changing tabs.

**Comment by tofumatt** [comment]
Ah right, I missed that! I've added a reset on `dimensionName` change, should be good now 👍🏻

**Review by felixarntz — CHANGES_REQUESTED** [review]
There still seems to be something off here, although I thought your code would fix it. When I run it, it only works sometimes.

Instead of relying on `useEffect`, wouldn't it be more reliable to unset it when switching the tab, together with the dimension? That would IMO also better clarify the intent here.

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/UserDimensionsPieChart.js [hunk 1]
```diff
@@ -84,6 +85,10 @@ export default function UserDimensionsPieChart( { dimensionName, entityURL, sour
 		const { chart, onSelect } = chartData || {};
 		const { slices } = UserDimensionsPieChart.chartOptions;
 
+		if ( selectedRow !== undefined ) {
+			chart.setSelection( [ { row: selectedRow } ] );
```
Comment (felixarntz) [inline assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/UserDimensionsPieChart.js:89]: I think the underlying problem here might be that we're selecting the `row` which is not specific enough, and it also shouldn't be necessary. We already store when a pie slice is selected, by storing the `dimensionValue` which _is_ the pie slice.

Basically, _if_ there is a (non-empty) `dimensionValue`, we need to ensure that the `row` which has that value as label (i.e. `row[ 0 ]` if I'm not mistaken) is selected.

**Review by felixarntz — CHANGES_REQUESTED** [review]
This is still not working as expected, it's unreliable and only appears to work on initial pageload. As soon as I do a couple more interactions (change date range, change tabs, back and forth etc) it doesn't keep the pie slice for the same dimension value selected like it should.

I left a comment below for what I think is the underlying problem here. I don't think we need to store a new piece of information actually, since `dimensionValue` already indicates which pie slice is/should be selected. Let's try to use that one.

**Comment by felixarntz** [comment]
Changed this to be against `master` since a fix for this issue should preferably go into the release.

**Comment by felixarntz** [comment]
Closing since this will be covered by #2702.

**Pull request closed without merging**